Automattic/jetpack
# Make the Flickr shortcode to be compatible with the last version of AMP

#### Changes proposed in this Pull Request:

* Make the Flickr shortcode to be compatible with the last version of AMP.
* The Flash code in the output for videos is replaced by a `<video>` tag.
* The fullscreen is removed for photos when an AMP page is showing.
* Two new params were added to the shortcode: `autoplay="yes"|"no"` (if the video plays automatically when the page is loaded) and `controls="yes"|"no"` (whether the video shows the controls)

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* This modifies the current Flickr shortcode output for video and photos.

#### Testing instructions:






1. Make sure the AMP plugin is activated.
2. Create a new post/page or edit an existing post/page
3. Add the Flickr shortcode to the content using the `video` param:
![](https://user-images.githubusercontent.com/51013565/82977838-aa609580-9fa8-11ea-9e6d-df57a4e556d7.jpg)
4. Save the post and check the output on the post frontend:
![](https://user-images.githubusercontent.com/51013565/82978295-d4ff1e00-9fa9-11ea-864d-52024d56c846.jpg)
5. Check output on AMP version:
![](https://user-images.githubusercontent.com/51013565/82978347-fc55eb00-9fa9-11ea-951b-12cb2f77f9b8.jpg)
6. Check the AMP validations errors by going to the Admin dashboard > AMP > Validated URLs and press the `Recheck` Button
![](https://user-images.githubusercontent.com/51013565/82979150-f52fdc80-9fab-11ea-8e0d-0209b0af9ad8.jpg)
7. Check that the Error table is empty.
8. Go back to the edit page and add a Flickr shortcode using the `photo` param:
![](https://user-images.githubusercontent.com/51013565/82978576-89993f80-9faa-11ea-9231-edbe77879381.jpg)
9. Save the post and check the post frontend:
![](https://user-images.githubusercontent.com/51013565/82978959-79ce2b00-9fab-11ea-8a2f-ef640ecf591a.jpg)
10. Check the AMP version:
![](https://user-images.githubusercontent.com/51013565/82979233-2a3c2f00-9fac-11ea-8748-fa539ab5f2e1.jpg)
11. Repeat the steps 6 and 7.
12. Please also check the combinations of params the shortcode has.



#### Proposed changelog entry for your changes:

* Improve AMP compatibility of the Flickr shortcode.
